docs/architecture/adr-081-protobuf-mgmt.md

```diff
@@ -0,0 +1,202 @@
+      the face of bigger changes.
+   2. Uploading our `.proto` files to Buf's registry on every release. This is
+      by far the most seamless for consumers of our `.proto` files, but requires
+      the dependency on Buf. This has the additional benefit that the Buf
+      registry will automatically [generate and host
+      documentation][buf-docs-gen] for these protos.
+   3. We could create a process that, upon release, creates a `.zip` file
+      containing our `.proto` files. This would require maintaining yet another
+      script/CI workflow that we would maintain.
+
+### Popular alternatives to Buf
+
+[Prototool] was not considered as it appears deprecated, and the ecosystem seems
+to be converging on Buf at this time.
+
+### Tooling complexity
+
+The more tools we have in our build/CI processes, the more complex and fragile
+repository/CI management becomes, and the longer it takes to onboard new team
+members. Maintainability is a core concern here.
+
+### Buf sustainability and costs
+
+One of the primary considerations regarding the usage of Buf is whether, for
+example, access to its registry will eventually become a
+paid-for/subscription-based service and whether this is valuable enough for us
+and the ecosystem to pay for such a service.
+
+Another consideration was Buf's sustainability as a project - what happens when
+their resources run out? Will there be a strong and broad enough open source
+community to continue maintaining it?
+
+### Local Buf usage options
+
+Local usage of Buf (i.e. not in CI) can be accomplished in two ways:
+
+1. Installing the relevant tools individually.
+2. By way of its [Docker image][buf-docker].
+
+Local installation of Buf requires developers to manually keep their toolchains
+up-to-date. The Docker option comes with a number of complexities, including
+how the file system permissions of code generated by a Docker container differ
+between platforms (e.g. on Linux, Buf-generated code ends up being owned by
+`root`).
+
+The trouble with the Docker-based approach is that we make use of the
+[gogoprotobuf] plugin for `protoc`. Continuing to use the Docker-based approach
+to using Buf will mean that we will have to continue building our own custom
+Docker image with embedded gogoprotobuf.
+
+Along these lines, we could eventually consider coming up with a [Nix]-based
+approach to developer tooling to ensure tooling consistency across the team and
+for anyone who wants to be able to contribute to Tendermint.
+
+## Decision
+
+We will adopt a hybrid approach for now that involves generating Go code using
+`protoc` locally (i.e. developers must manually generate Go code and commit them
+in their pull request submissions), but Buf for linting, breakage checking and
+its registry (mainly in CI, with optional usage locally).
+
+Failing CI when checking for breaking changes in `.proto` files will only happen
+when performing minor/patch releases.
```

Comment:
I somewhat worry that this will require us to do double duty when maintaining the protos since we need to make sure that the buf tooling works alongside the native protoc tooling. In effect, we've taken on the cost of buf while also taking on the cost of maintaining a homespun solution. If we are going to use buf for the registry and breakage checking, I'm not sure I see why we shouldn't just go all the way with it to avoid the hassle of maintaining parallel stacks.

Reply:
> In effect, we've taken on the cost of buf while also taking on the cost of maintaining a homespun solution.

Kind of. For now I haven't yet figured out how to configure buf to correctly generate the Go code without breaking the imports and package naming. By contrast, the equivalent code in a `Makefile` to generate what we want looks like [this](https://github.com/tendermint/tendermint/blob/d8b738f547f59aa991c9c0b7332d2a1fa52d0823/Makefile#L97).

Configuring buf for linting and breakage checking is trivial though.

Reply:
> Configuring buf for linting and breakage checking is trivial though.

If this is trivial, then I'm not worried :+1: